packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -135,157 +195,199 @@ describe("Presence", () => {
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
+
+						verifyAttendee(joinedAttendees[0], initialAttendeeConnectionId);
 					});
 
 					it('second time is announced once via `attendeeJoined` with status "Connected" even if most recent unknown to audience', () => {
 						// Setup - remove connection from audience
 						runtime.removeMember(rejoinAttendeeConnectionId);
 
 						// Act - simulate join message from client
-						presence.processSignal("", rejoinAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([rejoinAttendeeSignalInfo]);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
 
-						// Verify
-						verifyNewAttendee(rejoinAttendeeConnectionId);
+						verifyAttendee(joinedAttendees[0], rejoinAttendeeConnectionId);
 					});
 
 					it("as collateral and disconnected is NOT announced via `attendeeJoined`", () => {
 						// Setup - remove connections from audience
 						const collateralAttendeeConnectionId = "client3";
-						const collateralAttendeeSignal = generateBasicClientJoin(clock.now - 10, {
-							averageLatency: 40,
+						runtime.removeMember(initialAttendeeConnectionId);
+						runtime.removeMember(collateralAttendeeConnectionId);
+						const collateralAttendeeSignalInfo = {
 							clientSessionId: attendeeSessionId,
 							clientConnectionId: rejoinAttendeeConnectionId,
 							connectionOrder: 1,
+							fixedTime: clock.now - 10,
 							updateProviders: ["client2"],
 							priorClientToSessionId: {
-								...initialAttendeeSignal.content.data["system:presence"].clientToSessionId,
+								[initialAttendeeConnectionId]: {
+									rev: 0,
+									timestamp: initialAttendeeSignalInfo.fixedTime,
+									value: attendeeSessionId,
+								},
 								[collateralAttendeeConnectionId]: {
 									rev: 0,
 									timestamp: 0,
 									value: "collateral-id",
 								},
 							},
-						});
-						runtime.removeMember(initialAttendeeConnectionId);
-						runtime.removeMember(collateralAttendeeConnectionId);
+						};
 
 						// Act - simulate join message from client
-						presence.processSignal("", collateralAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([collateralAttendeeSignalInfo]);
 
-						// Verify only the rejoining attendee is announced
-						verifyNewAttendee(rejoinAttendeeConnectionId);
+						// Verify - only the rejoining attendee is announced
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
+
+						verifyAttendee(joinedAttendees[0], rejoinAttendeeConnectionId);
+					});
+
+					it.skip("is announced via `attendeeJoined` when a second joining attendee is unknown to audience", () => {
```

Comment:
I am not convinced of this test case.
I thought the case we cared about was not announcing a second-hand attendee. (But I didn't go back and look up the comment.)
And didn't the case involve this client not seeing the join for number two?
In any case the action here is in too large a step. initialAttendee is part of setup. The next two should be separately verified; only one of the two will generate a join and test should be firm about which one does.
Could defer all this by keeping the refactor to just a refactor and add test case second.

Reply:
Yes, you're right I rewrote it incorrectly; I've removed it from this PR and we can revisit it after to get it right

Reply:
The scenario you mention is actually already tested in "as collateral and disconnected is NOT announced via `attendeeJoined`". I believe the actual scenario we wanted to test had to do with connectivity, more specifically how old client connection information should be ignored (i.e. when a client receives a broadcast with old connection information, we should not set the attendee to disconnected based on audience check).